tensorly/decomposition/_nn_cp.py

```diff
@@ -291,6 +291,142 @@ def non_negative_parafac(tensor, rank, n_iter_max=100, init='svd', svd='numpy_sv
         return cp_tensor
 
 
+def non_negative_parafac_hals(tensor, rank, n_iter_max=100, init="svd", svd='numpy_svd', tol=1e-7,
+                              sparsity_coefficients=[], fixed_modes=[],hals='approx',
```

Comment:
Actually we used the same syntax as in _cp.py `parafac` function, which input `fixed_modes` has default [] (my bad since this was also from my older PR). We can change both `parafac` and `non_negative_parafac_hals` to have defaults `None` wherever we used `[]`.